c2rust-analyze/src/main.rs

```diff
@@ -150,6 +150,18 @@ fn run(tcx: TyCtxt) {
         gacx.fn_sigs.insert(ldid.to_def_id(), lsig);
     }
 
+    // Label the field types of each struct.
+    for ldid in tcx.hir_crate_items(()).definitions() {
+        let did = ldid.to_def_id();
+        if tcx.def_kind(did) != DefKind::Struct {
+            continue;
+        }
+        for field in tcx.adt_def(did).all_fields() {
+            let lty = gacx.assign_pointer_ids(tcx.type_of(field.did));
+            gacx.field_tys.insert(field.did, lty);
+        }
```

Comment:
Could you extract out a `gacx.label_field(field)` and `gacx.label_struct_fields(did)` method?  Not sure if it's that important, but I find it cleaner and easier to read/understand later.  Though this `run` function is quite long and I'd have the same comment for much of it, but I don't think it's worth it to change yet.